File: app/src/main/java/com/ipd/taxiu/ui/activity/coupon/IntegralExchangeActivity.java
package com.ipd.taxiu.ui.activity.coupon;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearSmoothScroller;
import android.support.v7.widget.RecyclerView;
import android.view.Menu;
import android.view.MenuItem;

import com.ipd.taxiu.R;
import com.ipd.taxiu.adapter.IntegralExchangeAdapter;
import com.ipd.taxiu.ui.BaseUIActivity;
import com.ipd.taxiu.ui.fragment.coupon.IntegralExchangeFragment;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Miss on 2018/7/30
 * 积分兑换
 */
public class IntegralExchangeActivity extends BaseUIActivity {
    @Override
    protected int getContentLayout() {
        return R.layout.activity_container;
    }

    @Override
    protected void initView(@Nullable Bundle bundle) {
        initToolbar();
    }

    @Override
    protected void loadData() {
       getSupportFragmentManager().beginTransaction().replace(R.id.fl_container, IntegralExchangeFragment.newInstance()).commit();
    }

    @Override
    protected void initListener() {

    }

    @NotNull
    @Override
    protected String getToolbarTitle() {
        return "积分兑换";
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_exchange_record,menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.integral_record){
            Intent intent = new Intent(this,ExchangeRecordActivity.class);
            startActivity(intent);

        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/ipd/rainbow/ui/activity/coupon/IntegralExchangeActivity.kt
package com.ipd.rainbow.ui.activity.coupon

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.ipd.rainbow.R
import com.ipd.rainbow.ui.BaseUIActivity
import com.ipd.rainbow.ui.fragment.coupon.IntegralExchangeFragment

/**
Created by Miss on 2018/8/10
积分兑换
 */
class IntegralExchangeActivity:BaseUIActivity() {
    companion object {
        fun launch(activity:Activity){
            val intent = Intent(activity,IntegralExchangeActivity::class.java)
            activity.startActivity(intent)
        }
    }

    override fun getContentLayout(): Int = R.layout.activity_container
    override fun getToolbarTitle(): String ="积分兑换"

    override fun initView(bundle: Bundle?) {
        initToolbar()
    }

    override fun loadData() {
        supportFragmentManager.beginTransaction().replace(R.id.fl_container,IntegralExchangeFragment.newInstance()).commit()
    }

    override fun initListener() {
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_exchange_record,menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        val id:Int = item!!.itemId
        if (id == R.id.integral_record){
            ExchangeRecordActivity.launch(this)
        }
        return super.onOptionsItemSelected(item)
    }
}